app/models/action.rb

```diff
@@ -3,5 +3,13 @@ class Action < ActiveRecord::Base
   belongs_to :member
 
   has_paper_trail on: [:update, :destroy]
+
+  after_create :update_member_donor_status
```

Comment:
I agree with your point, but the problems I have are as follows:

* tightly couples action (creation of) with member.
* confusing for new developers to the codebase. Something to member is happening without any context. By having the member update its state within the payments controller, or somewhere in the payment workflow, the narrative become very clear to the code reader.
* cat be hard to unit test when a model has side effects like this.
* encourages new contributors to repeat pattern.